src/main/java/org/mule/module/apikit/validation/attributes/HeadersValidator.java

```diff
@@ -164,8 +164,10 @@ private static List<String> deserializeValues(List<String> listOfDelimitedValues
   }
 
   private static void validateTypeArrayValues(String name, List<String> values, Parameter type) throws InvalidHeaderException {
-    String yamlArrayValue = values.stream().collect(Collectors.joining("\n- ", "- ", ""));
-    validateTypeValue(name, yamlArrayValue, type);
+    if (!type.validateArray(values)) {
+      throw new InvalidHeaderException(format("Invalid values '%s' for header '%s'", values.stream().collect(joining(",")),
```

Comment:
Probably a bit clearer:
```suggestion
      throw new InvalidHeaderException(format("Invalid values '%s' for header '%s'", join(",", values)),
```

In that case the static import for `java.util.stream.Collectors.joining` should be removed and the static import for `java.lang.String.join` should be added.